### ISSUE TITLE
Compilation error in KotlinGenerator generated DAOs for data type TINYINT

### ISSUE THREAD

**Issue by davidkarlsen** [jOOQ/jOOQ#10779]
Related to https://github.com/jOOQ/jOOQ/issues/10778

the following is generated for the flyway version schema:
```
    fun fetchBySuccess(vararg values: Byte): List<com.edb.fs.tac.jfr.srv.jooq.tables.pojos.SchemaVersion> = fetch(SchemaVersion.SCHEMA_VERSION.SUCCESS, *values)
```
which leads to a type mismatch:
Required: Array<(out) Byte?>!
Found: ByteArray

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'm assuming this is in your generated `DAO` class? I can't reproduce this with an ordinary `BINARY` type. Can you please post the DDL for the `schema_version` table?

**Comment by davidkarlsen** [comment]
```sql
  CREATE TABLE "APPDATA"."schema_version" 
   (	"installed_rank" NUMBER(*,0) NOT NULL ENABLE, 
	"version" VARCHAR2(50 BYTE), 
	"description" VARCHAR2(200 BYTE) NOT NULL ENABLE, 
	"type" VARCHAR2(20 BYTE) NOT NULL ENABLE, 
	"script" VARCHAR2(1000 BYTE) NOT NULL ENABLE, 
	"checksum" NUMBER(*,0), 
	"installed_by" VARCHAR2(100 BYTE) NOT NULL ENABLE, 
	"installed_on" TIMESTAMP (6) DEFAULT CURRENT_TIMESTAMP NOT NULL ENABLE, 
	"execution_time" NUMBER(*,0) NOT NULL ENABLE, 
	"success" NUMBER(1,0) NOT NULL ENABLE, 
	 CONSTRAINT "schema_version_pk" PRIMARY KEY ("installed_rank")
  USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "APP_DATA"  ENABLE
   ) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "APP_DATA" ;

  CREATE INDEX "APPDATA"."schema_version_s_idx" ON "APPDATA"."schema_version" ("success") 
  PCTFREE 10 INITRANS 2 MAXTRANS 255 COMPUTE STATISTICS 
  STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "APP_DATA" ;
```

**Comment by lukaseder** [comment]
Thanks a lot. I can reproduce it with that table definition. Will investigate now. (Thanks for all your reports, by the way, that's very helpful!)

**Comment by lukaseder** [comment]
We need to append `*values.toTypedArray()` as we already do for all the other primitive wrapper types. Byte was just forgotten in the relevant config.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0. Backport scheduled for 3.14.1 (#10786)
